packages/flutter/lib/src/services/deferred_component.dart

```diff
@@ -0,0 +1,157 @@
+  /// library loading process. For example:
+  ///
+  /// ```dart
+  /// import 'split_module.dart' deferred as SplitModule;
+  /// ...
+  /// SplitModule.loadLibrary();
+  /// ```
+  ///
+  /// This method will not load associated dart libraries contained in the dynamic
+  /// feature module, though it will download the files necessary and subsequent
+  /// calls to `loadLibrary()` to load will complete faster.
+  static Future<void> installDeferredComponent({@required String? moduleName}) async {
+    await SystemChannels.deferredComponent.invokeMethod<void>(
+      'installDeferredComponent',
+      <String, dynamic>{ 'loadingUnitId': -1, 'moduleName': moduleName },
+    );
+  }
+  /// Gets the current installation state of the deferred component identified by the
+  /// [moduleName].
+  ///
+  /// This method returns a string that represents the state. Depending on
+  /// the implementation, this string may vary, but the default Google Play
+  /// Store implementation retuns a state in the [DeferredComponentInstallState] enum.
+  ///
+  /// Installations of deferred component modules may be triggered by either calling
+  /// [installDeferredComponent] for assets-only modules or `loadLibrary()` on a deferred
+  /// imported library. Modules not yet requested or do not exist will complete with
+  /// [DeferredComponentInstallState.unknown]. Modules previously installed but not
+  /// loaded in this session will return [DeferredComponentInstallState.installedPendingLoad],
+  /// which indicates that the installtion request call should be repeated to
+  /// complete the loading process.
+  ///
+  /// Due to the async nature of platform channels and network i/o, newly requested
+  /// installs may return null until the installation request has been processed. Thus,
+  /// this state information should be used as purely informational and not as an
+  /// accurate reflection of the readiness of the deferred component. Code and assets
+  /// may only be used after the Future completes succesfully, regardless of what
+  /// the installation state is in.
+  static Future<String?> getDeferredComponentInstallState({@required String? moduleName}) async {
+    await SystemChannels.deferredComponent.invokeMethod<void>(
+      'getDeferredComponentInstallState',
+      <String, dynamic>{ 'loadingUnitId': -1, 'moduleName': moduleName },
+    );
+  }
+
+  /// Requests that a deferred component identified by the [moduleName] be
+  /// uninstalled.
+  ///
+  /// Since uninstallation typically requires significant disk i/o, this method only
+  /// signals the intent to uninstall. Actual uninstallation (eg, removal of
+  /// assets and files) may occur at a later time. However, once uninstallation
+  /// is requested, the deferred component should not be used anymore until
+  /// [installDeferredComponent] or `loadLibrary()` is called again.
+  static Future<void> uninstallDeferredComponent({@required String? moduleName}) async {
```

Comment:
nit: maybe add a "see also" section here and to installDeferredComponent which links to the other?